pkg/sloop/processing/relationship.go

```diff
@@ -0,0 +1,126 @@
+package processing
+
+import (
+	"encoding/json"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/pkg/errors"
+	"github.com/salesforce/sloop/pkg/sloop/kubeextractor"
+	"github.com/salesforce/sloop/pkg/sloop/store/typed"
+	"github.com/salesforce/sloop/pkg/sloop/store/untyped"
+	"github.com/salesforce/sloop/pkg/sloop/store/untyped/badgerwrap"
+)
+
+
+
+
+//  transfer watchRec.Payload to Relationship payload: Convert watchRec to record in relationship table and save key-value pair
+func updateRelationshipTable(tables typed.Tables, txn badgerwrap.Txn, watchRec *typed.KubeWatchResult, metadata *kubeextractor.KubeMetadata) error {
+	if watchRec.Kind != kubeextractor.PodKind {
+		glog.Infof("Skipping relationship table update as kubewatch result is not PodKind")
+		return nil
+	}
+	// convert node -> pod relationship from watchRec
+	// get new relationship record from watch record
+	record, err := getRelationshipRec(watchRec)
+
+	//convert timestamp
+	ts, err := ptypes.Timestamp(watchRec.Timestamp)
+	if err != nil {
+		return errors.Wrap(err, "could not convert timestamp")
```

Comment:
consider adding the bad timestamp in the error message

Reply:
updated